dag/src/lib.rs

```diff
@@ -137,31 +166,47 @@ impl Edge {
 pub struct Node {
     entry: Edge,
     template_name: String,
+    parameter_instances: Vec<BigInt>,
```

Comment:
Each node in the DAG now stores the values of the parameters used to instantiate it